crates/ra_assists/src/assists/add_explicit_type.rs

```diff
@@ -25,31 +30,85 @@ pub(crate) fn add_explicit_type(ctx: AssistCtx<impl HirDatabase>) -> Option<Assi
     let stmt = ctx.find_node_at_offset::<LetStmt>()?;
```

Comment:
I think let and `_` cases are sufficiently different that there's little sense to handle both in the same function. There are cases like `xs.iter().collect::<_>()`, where there are no `let` at all. 

I think we should add a second `pub(crate)` function in this file

```rust
pub(crate) fn fill_placeholder_type(ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
   let placeholder = cts.find_node_at_offset::<ast::Placeholder>()?;
   ...

```